gradle-idea-configuration/src/main/groovy/com/palantir/gradle/ideaconfiguration/ConfigureXml.groovy

```diff
@@ -17,27 +17,27 @@
 package com.palantir.gradle.ideaconfiguration
 
 class ConfigureXml {
```

Comment:
FLUP: We shouldn't use Groovy at all - it has no type checking. We should do this in Java instead. Even for the interim period when rolling this out where we want to still read the old values, we should read them using a Java XmlMapper

Particularly, we want the source set to be `java`. Groovy source sets are bad as they do not run errorprones which would have caught some issues below

Reply:
TODO: make a gradle-guide check for groovy source sets